### PR TITLE
Add radial weighting for near axis fit

### LINKED ISSUE #798: Add radial weighting when fitting to near axis
We trust the nae closest to the axis, so doing a weighted fit with weight like 1/rho or something would likely give better results than uniform weighting, especially at low aspect ratio

### PR DESCRIPTION
**Pull request by f0uriest**
By construction, the near axis expansion is most accurate near the axis, so this PR adds a radial weighting when we fit an equilibrium from a near axis solution. Default is `1/rho**2` which seems to work well on a few equilibria I tried.

Resolves #798

### CODE REVIEW

File: desc/equilibrium/equilibrium.py [hunk 1]
```diff
@@ -1586,7 +1586,15 @@ def resolution_summary(self):
...
-        cls, na_eq, r=0.1, L=None, M=8, N=None, ntheta=None, spectral_indexing="ansi"
+        cls,
+        na_eq,
+        r=0.1,
+        L=None,
+        M=8,
+        N=None,
+        ntheta=None,
+        spectral_indexing="ansi",
+        w=2,
```
Comment (ddudt) [inline desc/equilibrium/equilibrium.py:1597]: In theory, should this be related to the "order" of the NAE constraints we use? If so, I would recommend changing the variable name to `order` or whatever we use in those NAE constraint functions.

Comment (f0uriest) [reply desc/equilibrium/equilibrium.py:1597]: I don't think theres a direct relation. While the order of the expansion usually means the distance from the axis that we trust it, the accuracy also depends on the shape of the axis and r/r_crit etc. In all cases we know the axis is "most accurate" though the exact degree of falloff isn't clear

Comment (dpanici) [reply desc/equilibrium/equilibrium.py:1597]: I agree, the order of the NAE would not relate to the weighting near-axis that we are trying to do here.

**Review by dpanici — APPROVED** [review]
Not sure if I will notice much of a difference but is good to have. re-trying my paper results to see if anything changes, and would be very cool if this can let us, with agressive weights, fit NAE surfaces that are very crappy at rho=1 with more reasonable surfaces so our initial guess for NAE constraint isnt so bad

**Review by ddudt — APPROVED** [review]

**Pull request merged**